feat: signout api integration

sidarth-23 opened the pull request:
<!--
  Thanks for submitting a pull request!
  We appreciate you spending the time to work on these changes. Please provide enough information so that others can review your pull request. 
-->

## Summary
Refactored the cookieDestroyer function to send a POST request to the LOGOUT_URL with the refresh token and authorization header.  Upon successful logout, the function now deletes the "Authorization" and "Refresh" cookies and redirects to "/home".

<!--
 Explain the **motivation** of this Pull Request
-->

- [x] Tested (Must)
- [ ] Test Case added
- [x] Build Successful (Must)
- [x] Sufficient Code comments added (Must)
- [ ] Attached Screenshots / Videos <!-- if the PR contains UI changed, fixes, improvements -->
- [x] All Relevant Documents added


## Depends on

<!--- Does this PR depend on another PR that should be merged first or at the same time -->

yogai53 commented on apps/admin/actions/cookie-destroy.ts:
```diff
@@ -1,10 +1,28 @@
+    redirect("/home");
+  } catch (e) {
+    // TODO: Handle error
+    console.error(e);
```
Should we need to send some error?

sidarth-23 replied:
I'm thinking we need to prevent user logout if the api fails to make sure the refresh token is dealt with

yogai53 replied:
Yes. But we should show some alert?

yogai53 commented on apps/admin/actions/cookie-destroy.ts:
```diff
@@ -1,10 +1,28 @@
+    );
+    cookies().delete("Authorization");
+    cookies().delete("Refresh");
+    redirect("/home");
```
Why home? Shoudn't it be back to login page?

sidarth-23 replied:
Yep. That was meant to be the route after middleware setup. Also fixed it in cookie setter

sidarth-23 replied:
For the time being we are leaving logged out users in the starting page. But I think we will take care of the minor details sometime later

sidarth-23 committed:
feat: signout api integration

Refactored the cookieDestroyer function to send a POST request to the LOGOUT_URL with the refresh token and authorization header. 
Upon successful logout, the function now deletes the "Authorization" and "Refresh" cookies and redirects to "/home".
feat: Add logout URL to authentication endpoints

Added a new constant for the logout URL in the authentication endpoints to 
handle user sign out functionality.
feat: Update redirect paths in cookieSetter and cookieDestroyer functions

Changed redirect paths in cookieSetter from "/admin/dashboard" to "/dashboard" and in cookieDestroyer from "/home" to "/". Added handling error message to prevent signout and token exposure.

sidarth-23 committed:
Merge branch 'master' into feat/signout-api-integration
